src/versit/qversitcontactimporter_p.cpp

```diff
@@ -983,26 +985,87 @@ QString QVersitContactImporterPrivate::takeFirst(QList<QString>& list) const
+    QString groupMonth = QStringLiteral("month");
+    QString groupYear = QStringLiteral("year");
+    QString groupHr = QStringLiteral("HH");
+    QString groupMin = QStringLiteral("mm");
+    QString groupSec = QStringLiteral("ss");
+    QString dayPattern = QStringLiteral("(?<%1>\\d{1,2})").arg(groupDay);
+    QString monthPattern = QStringLiteral("(?<%1>\\d{1,2})").arg(groupMonth);
+    QString yearPattern = QStringLiteral("(?<%1>\\d{4}|%2)").arg(groupYear, separatorPattern);
+    QString timePattern = QStringLiteral("T(?<%1>\\d{1,2})%4(?<%2>\\d{1,2})%4(?<%3>\\d{1,2})").arg(groupHr, groupMin,
+                                                                                                   groupSec, QStringLiteral("(-?|[:])"));
+    QString regexShortDatePattern = hasSeparator ? QStringLiteral("^%1") % separatorPattern % QStringLiteral("%2")
+                                                 : QStringLiteral("^%1%2");
+    QString regexFullDatePattern = QStringLiteral("%1%2").arg(regexShortDatePattern, hasSeparator ? separatorPattern % QStringLiteral("%3")
+                                                                                                  : QStringLiteral("%3"));
+    QDateTime dateTime;
+    QRegularExpression re;
+    QRegularExpressionMatch match;
+
     if (utc)
         value.chop(1); // take away z from end;
 
-    QDateTime dateTime;
-    if (value.contains(QLatin1Char('-'))) {
-        dateTime = QDateTime::fromString(value,Qt::ISODate);
-        hasTime = dateTime.isValid() && value.contains(QLatin1Char('T'));
-    } else {
-        switch (value.length()) {
-        case 8:
-            dateTime = QDateTime::fromString(value, QStringLiteral("yyyyMMdd"));
-            break;
-        case 15:
-            dateTime = QDateTime::fromString(value, QStringLiteral("yyyyMMddThhmmss"));
-            hasTime = true;
-            break;
-        // default: return invalid
+    // add time mask
+    if (hasTimeKey) {
+        regexShortDatePattern.append(timePattern);
+        regexFullDatePattern.append(timePattern);
+    }
+
+    re.setPattern(regexFullDatePattern.arg(yearPattern, monthPattern, dayPattern));
+    match = re.match(value);
+
+    // if the date is not in ISO format
+    if (!match.hasMatch()) {
+        re.setPattern(regexFullDatePattern.arg(dayPattern, monthPattern, yearPattern));
+        match = re.match(value);
+        // if year not specified in the date
+        if (!match.hasMatch()) {
+            re.setPattern(regexShortDatePattern.arg(monthPattern, dayPattern));
+            match = re.match(value);
+        }
+    }
+
+    if (match.hasMatch()) {
+        auto checkDateFormat = [&](const QString &number, bool isYear = false) {
+            if (!isYear) {
+                // converting to ISO standard month and day
+                return number.length() < 2 ? "0" % number : number;
+            } else {
+                // setting the current year if it is missing from the date
```

Comment:
Corner case but in case the date is --0229 it will be accepted or not depending on the current year :) Then again the current year is probably least surprising to the user if the year gets shown.

Reply:
Such a case will not be processed now and the date will not be displayed.